eth_verifier/lib/msgpack/Deserialize.sol

```diff
@@ -57,6 +60,24 @@ library MsgPk {
         self.curr_index += n;
     }
 
+    error EncodedMapKeyNotFound();
+
+    function find_value(EncodedMap memory self, string memory key)
+        public
+        pure
+        returns (bytes memory)
+    {
+        uint256 i = 0;
+        while (
+            i != self.keys.length &&
+            keccak256(bytes(self.keys[i])) != keccak256(bytes(key))
+        ) {
```

Comment:
this should be changed by an array of elements already hashed, or we sould apply the Gas optimization showed here: https://www.educative.io/answers/how-to-compare-two-strings-in-solidity

In the other hand, this function call should be done only once, outside the loop:
```
keccak256(bytes(key)
```

Reply:
Seems like a good idea to make the `EncodedMap` struct more like a hash map as it should be. The reason of not using a `mapping` is because it can exist in sotrage memory only. Maybe there's already a library out there that implements memory `mappings` and an efficient string comparison function.

Should I take care of this now or open an issue?

Reply:
We decided that it's best to leave this as is and open an issue to resolve it later.